src/transformers/models/cogvlm/processing_cogvlm.py

```diff
@@ -0,0 +1,149 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Processor class for CogVLM.
+"""
+
+from typing import List, Optional, Union
+
+from ...image_processing_utils import BatchFeature
+from ...image_utils import ImageInput
+from ...processing_utils import ProcessorMixin
+from ...tokenization_utils_base import PaddingStrategy, PreTokenizedInput, TextInput, TruncationStrategy
+from ...utils import TensorType
+
+
+LANGUAGE_TOKEN_TYPE = 0
+VISION_TOKEN_TYPE = 1
+
+
+class CogVLMProcessor(ProcessorMixin):
+    r"""
+    Constructs a CogVLM processor which wraps a CLIP image processor and a LLaMa tokenizer into a single processor.
+
+    [`CogVLMProcessor`] offers all the functionalities of [`CLIPImageProcessor`] and [`LlamaTokenizer`]. See the docstring
+    of [`~CogVLMProcessor.__call__`] and [`~CogVLMProcessor.decode`] for more information.
+
+    Args:
+        image_processor (`CLIPImageProcessor`):
+            An instance of [`CLIPImageProcessor`]. The image processor is a required input.
+        tokenizer (`AutoTokenizer`):
+            An instance of ['LlamaTokenizer`]. The tokenizer is a required input.
+    """
+
+    attributes = ["image_processor", "tokenizer"]
+    image_processor_class = "CLIPImageProcessor"
+    tokenizer_class = "LlamaTokenizer"
+
+    def __init__(self, image_processor, tokenizer, image_size, patch_size):
+        super().__init__(image_processor, tokenizer)
+        self.image_size = image_size
+        self.patch_size = patch_size
```

Comment:
cc @ydshieh for this model, I need to store 2 attributes to the processor, however we currently don't have a `processor_config.json` file. Can we add support for this in `from_pretrained` and `save_pretrained`?